cadc-dali/src/main/java/ca/nrc/cadc/dali/tables/parquet/ParquetWriter.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    private static OutputFile outputFileFromStream(OutputStream outputStream) {
+        return new OutputFile() {
+            @Override
+            public PositionOutputStream create(long blockSizeHint) {
+                // TODO: use blockSizeHint argument probably assigning to OutputStream.
+                return new PositionOutputStream() {
+                    private long position = 0;
+
+                    @Override
+                    public void write(int b) {
+                        try {
+                            outputStream.write(b);
+                            position++;
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+
+                    @Override
+                    public void write(byte[] b) {
+                        try {
+                            outputStream.write(b);
+                            position += b.length;
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+
+                    @Override
+                    public void write(byte[] b, int off, int len) {
+                        try {
+                            outputStream.write(b, off, len);
+                            position += len;
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+
+                    @Override
+                    public long getPos() {
+                        return position;
+                    }
+
+                    @Override
+                    public void close() {
+                        try {
+                            outputStream.close();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                };
+            }
+
+            @Override
+            public PositionOutputStream createOrOverwrite(long blockSizeHint) {
+                return create(blockSizeHint);
+            }
+
+            @Override
+            public boolean supportsBlockSize() {
+                return false;
+            }
+
+            @Override
+            public long defaultBlockSize() {
+                return 0;
+            }
+        };
+    }
+
+    private static void updateVOTable(VOTableResource voTableResource) {
+        List<VOTableField> fields = voTableResource.getTable().getFields();
+
+        // Convert timestamp fields to long to ensure inconsistencies between avro schema field and votable metadata field
+        for (int i = 0; i < fields.size(); i++) {
+            VOTableField field = fields.get(i);
+
+            if ("timestamp".equals(field.xtype)) {
+                fields.set(i, new VOTableField(field.getName(), "long"));
```

Comment:
Yes, the VOTableField datatype is immutable so you have to create a new one.. but there are a bunch of optional metadata in the source field as well (aka public fields that can be null): some of that is still applicable and some is not.

In the case of timestamp (aside from the datatype,arraysize,xtype change): the units could be set explicitly to `msec` (I think - you will need to check the VOUnits standard) but I think everything else is still applicable.

In the case of short below, all the optional metadata is still applicable.